### PR TITLE
[SYCLomatic] Support migration of make_half2.

### PR DESCRIPTION
**Pull request by tangjj11**
Signed-off-by: Tang, Jiajun jiajun.tang@intel.com

### CODE REVIEW

**Review by yihwang-nv — APPROVED** [review]

File: clang/test/dpct/math/half/half_cuda11_after.cu [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+#include "cuda_fp16.h"
+
+__global__ void kernelFuncHalfConversion() {
+  half h;
+  half2 h2;
+  // CHECK: h2 = sycl::half2(h, h);
+  h2 = make_half2(h, h);
+}
+
+int main() { return 0; }
```
Comment (yihwang-nv) [inline clang/test/dpct/math/half/half_cuda11_after.cu]: it can be removed

File: clang/test/dpct/math/half/half_ext_cuda11_after.cu [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+#include "cuda_fp16.h"
+
+__global__ void kernelFuncHalfConversion() {
+  half h;
+  half2 h2;
+  // CHECK: h2 = sycl::half2(h, h);
+  h2 = make_half2(h, h);
+}
+
+int main() { return 0; }
```
Comment (yihwang-nv) [inline clang/test/dpct/math/half/half_ext_cuda11_after.cu]: Same as above

**Review by wenhuiNi — APPROVED** [review]

File: clang/test/dpct/math/half/half_cuda11_after.cu (hunk 1 above)
Comment (tangjj11) [reply clang/test/dpct/math/half/half_cuda11_after.cu]: add the calling to kernel in main.

File: clang/test/dpct/math/half/half_ext_cuda11_after.cu (hunk 2 above)
Comment (tangjj11) [reply clang/test/dpct/math/half/half_ext_cuda11_after.cu]: add the calling to kernel in main.

**Review by ziranzha — APPROVED** [review]

**Pull request merged**